File: ruby/easy/longest_continuous_increasing_subsequence.rb
# @param {Integer[]} nums
# @return {Integer}
def find_length_of_lcis(nums)
    return 0 if nums.length == 0
    lcis = count = 1
    for i in 1..nums.length-1
        if(nums[i] > nums[i-1])
            count += 1
        else
            count = 1
        end
        lcis = [lcis, count].max
    end
    return lcis
end

File: javascript/easy/longestContinuousIncreasingSubsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLengthOfLCIS = function(nums) {
    let lcis = 0, i = 0, count = 0;
    while(i < nums.length) {
        if(i === 0 || nums[i] > nums[i-1]) {
            count++;
        } else {
            count = 1;
        }
        lcis = Math.max(count, lcis);
        i++;
    }
    return lcis;
};
